docs/decisions/0006-middleware-support-for-http-clients.rst

```diff
@@ -0,0 +1,45 @@
+Middleware Support for HTTP clients
+===================================
+
+Status
+------
+
+Proposed
```

Comment:
Accepted upon merge :)

Reply:
May just be worth setting to "Accepted" in the PR here so that it's right once it hits master. :D